File: projects/10/ruby/birds.rb
# birds.rb | Tests out the menagerie of bird classes.
#
# Begun by: Dr. Adams, for CS 214 at Calvin College.
## Completed by: Justin Voss
## Date: 03/24/22
######################################################

require './Bird.rb'
require './duck.rb'
require './goose.rb'
require './owl.rb'
require './ostrich.rb'
require './penguin.rb'
require './kiwi.rb'
require './FlyingBird.rb'
require './WalkingBird.rb'

puts "\nWelcome to the Bird Park!\n\n"

bird0 = Bird.new "Hawkeye"
bird0.print

bird1 = Duck.new "Donald"
bird1.print

bird2 = Goose.new "Mother"
bird2.print

bird3 = Owl.new "Woodsey"
bird3.print

bird4 = Ostrich.new "Ozzie"
bird4.print

bird5 = Penguin.new "Pingu"
bird5.print

bird6 = Kiwi.new "Ninja"
bird6.print

puts "\n\n"

File: projects/10/java/Birds.java
/* Birds.java illustrates inheritance and polymorphism.
 *
 * Begun by: Joel Adams, for CS 214 Lab 10 at Calvin College.
 * Completed by: Justin Voss
 * Date: 03/24/2022
 *********************************************************/

public class Birds {
  
  public static void main(String[] args) {

      System.out.println("\nWelcome to the Bird Park!\n");

      Duck bird2 = new Duck("Donald");
      bird2.print();

      Goose bird3 = new Goose("Mother Goose");
      bird3.print();

      Owl bird4 = new Owl("Woodsey");
      bird4.print();

      Ostrich bird5 = new Ostrich("Oliver");
      bird5.print();

      Penguin bird6 = new Penguin("Pingu");
      bird6.print();

      Kiwi bird7 = new Kiwi("Ninja");
      bird7.print();

      System.out.println();
    }
}

